vortex-array/src/implementation.rs

```diff
@@ -207,14 +207,14 @@ impl<T: AsArray> ArrayStatistics for T {
     }
 }
 
-impl<D> From<D> for ArrayData
+impl<D> ToArrayData for D
 where
-    D: IntoArray + ArrayEncodingRef + ArrayStatistics + GetArrayMetadata,
+    D: IntoArray + ArrayEncodingRef + ArrayStatistics + GetArrayMetadata + Clone,
 {
-    fn from(value: D) -> Self {
+    fn to_array_data(&self) -> ArrayData {
         // TODO: move metadata call `Array::View` match
-        let metadata = value.metadata();
-        let array = value.into_array();
+        let metadata = self.metadata();
```

Comment:
for future reference - the general issue is that `ArrayView` alone doesn't have the type information needed to deserialize metadata here, so a more generic `From` implementation just doesn't work.